joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1425,9 +1411,29 @@ bool JointTrajectoryController::validate_trajectory_msg(
     }
   }
 
+  // CHECK: if trajectory end time is in the past (if start time defined)
+  const auto trajectory_start_time = static_cast<rclcpp::Time>(trajectory.header.stamp);
```

Comment:
```suggestion
  const rclcpp::Time trajectory_start_time = trajectory.header.stamp;
```

I don't know if this will work, but I'm guessing yes from here

https://github.com/ros2/rclcpp/blob/19773973a80d753c6fa028b0b548462fbbef122d/rclcpp/src/rclcpp/time.cpp#L109-L114